core-primitives/enclave-metrics/src/lib.rs

```diff
@@ -24,11 +24,24 @@ compile_error!("feature \"std\" and feature \"sgx\" cannot be enabled at the sam
 extern crate sgx_tstd as std;
 
 use codec::{Decode, Encode};
+use std::string::String;
+use substrate_fixed::types::U32F32;
 
-#[derive(Encode, Decode)]
+#[derive(Encode, Decode, Debug)]
 pub enum EnclaveMetric {
 	SetSidechainBlockHeight(u64),
 	TopPoolSizeSet(u64),
 	TopPoolSizeIncrement,
 	TopPoolSizeDecrement,
+	ExchangeRateOracle(ExchangeRateOracleMetric),
+}
+
+#[derive(Encode, Decode, Debug)]
+pub enum ExchangeRateOracleMetric {
+	/// Exchange Rate from CoinGecko - (Source, TradingPair, ExchangeRate)
+	ExchangeRate(String, String, U32F32),
+	/// Response time of the request in [ms]. (Source, ResponseTime)
+	ResponseTime(String, u128),
+	/// Increment the number of requests (Source)
+	NumberRequestsIncrement(String),
```

Comment:
Extension for the metrics data type for the oracle

Reply:
In the issues it's state to monitor
  - teeracle enclave balance alerts
  -  extrinsic failures
  -  general service health
  
 How does this match the implemented metrics?

Reply:
These here are the metrics from inside the enclave. All the metrics you mention originate from the untrusted service and therefore are not part of the `enclave-metrics`. See https://github.com/integritee-network/teeracle/pull/22/files#r830842046.
The general service health metrics (uptime, number of threads, memory usage etc.) are part of the Prometheus base metrics, so they don't need to be defined explicitly.

Reply:
I see. So `EXTRINSIC_INCLUSION_SUCCESS` maps to `extrinsic failures`, and the balance is covered here: https://github.com/integritee-network/teeracle/blob/ae64a69ec4c40f785ce8bbafd7b0e70c3c9c3d3f/service/src/prometheus_metrics.rs#L112

Thanks :)